jumanji/environments/routing/macvrp/env_test.py

```diff
@@ -0,0 +1,269 @@
+
+from jumanji.environments.routing.macvrp.constants import DEPOT_IDX
+from jumanji.environments.routing.macvrp.env import MACVRP, Observation
+from jumanji.environments.routing.macvrp.test_data import (
+    test_action_mask,
+    test_node_demand,
+)
+from jumanji.environments.routing.macvrp.types import State
+from jumanji.testing.env_not_smoke import check_env_does_not_smoke
+from jumanji.testing.pytrees import assert_is_jax_array_tree
+from jumanji.types import TimeStep
+
+
+class TestEnvironmentSpec:
+    def test_macvrp__reset(self, macvrp_env: MACVRP) -> None:
+        """Validates the jitted reset of the environment."""
+        chex.clear_trace_counter()
+        reset_fn = jax.jit(chex.assert_max_traces(macvrp_env.reset, n=1))
+
+        key = jax.random.PRNGKey(0)
+        state, timestep = reset_fn(key)
+
+        # # Call again to check it does not compile twice.
+        state, timestep = reset_fn(key)
+
+        assert isinstance(timestep, TimeStep)
+        assert isinstance(state, State)
+
+        # Initial position is at depot, so current capacity is max capacity
+        assert jax.numpy.all(state.vehicles.capacities == macvrp_env._max_capacity)
+
+        # # All vechicles are at the depot
+        assert jax.numpy.all(state.vehicles.positions == DEPOT_IDX)
+
+        # Cost of the depot must be 0
+        assert state.nodes.demands[DEPOT_IDX] == 0
+
+        # Check that the time windows are valid.
+        assert jax.numpy.all(0 < state.windows.start)
+        assert jax.numpy.all(state.windows.start < macvrp_env._max_start_window)
+        assert jax.numpy.all(state.windows.end > state.windows.start)
+
+        # Check that the early_coefs and late_coefs are valid.
+        assert jax.numpy.all(macvrp_env._early_coef_rand[0] <= state.coeffs.early)
+        assert jax.numpy.all(state.coeffs.early < macvrp_env._early_coef_rand[1])
+        assert jax.numpy.all(macvrp_env._late_coef_rand[0] <= state.coeffs.late)
+        assert jax.numpy.all(state.coeffs.late < macvrp_env._late_coef_rand[1])
+
+        assert_is_jax_array_tree(state)
+
+    def test_macvrp__step(self, macvrp_env: MACVRP) -> None:
+        """Validates the jitted step of the environment."""
+        chex.clear_trace_counter()
+
+        key = jax.random.PRNGKey(0)
+        reset_fn = jax.jit(chex.assert_max_traces(macvrp_env.reset, n=1))
+        state, timestep = reset_fn(key)
+
+        # Check the timestep step type is FIRST
+        assert timestep.first()
+        # Starting position is depot, new action to visit first node
+        new_actions = jax.numpy.array(
+            jax.numpy.arange(1, macvrp_env._num_vehicles + 1), dtype=np.int16
+        )
+
+        step_fn = jax.jit(chex.assert_max_traces(macvrp_env.step, n=1))
+        new_state, next_timestep = step_fn(state, new_actions)
+
+        # Check that the state has changed
+        assert not jax.numpy.array_equal(
+            new_state.vehicles.time_penalties, state.vehicles.distances
```

Comment:
Why checking the time_penalties against the distances ?

Reply:
Good catch thanks! Updated.